src/parser/jobs/pld/modules/Requiescat.tsx

```diff
@@ -23,223 +19,109 @@ const SEVERITIES = {
-					const epochTimestamp = this.parser.fflogsToEpoch(timestamp)
-					reqState.isRushing = false
-						|| this.invulnerability.isActive({timestamp: epochTimestamp, types: ['invulnerable']})
-						|| this.invulnerability.isActive({timestamp: epochTimestamp, types: ['untargetable']})
-				})
-			}
-
-			this.requiescats.push(reqState)
-		}
+	override initialise() {
+		super.initialise()
+
+		this.addEvaluator(new ExpectedActionsEvaluator({
+			expectedActions: [
+				{action: [this.data.actions.HOLY_SPIRIT, this.data.actions.HOLY_CIRCLE], expectedPerWindow: 3},
+				{action: this.data.actions.CONFITEOR, expectedPerWindow: 1},
+			],
+			suggestionIcon: this.data.actions.HOLY_SPIRIT.icon,
+			suggestionContent: <Trans id="pld.requiescat.suggestions.wrong-gcd.content">
+				GCDs used during <DataLink action="REQUIESCAT" /> should consist of 3-4 uses of <DataLink action="HOLY_SPIRIT" /> (or
+				multi-hit <DataLink action="HOLY_CIRCLE" />) and 1 use of <DataLink action="CONFITEOR" /> for optimal damage.
+			</Trans>,
+			suggestionWindowName: <DataLink action="REQUIESCAT" showIcon={false} />,
+			severityTiers: {
+				1: SEVERITY.MEDIUM,
+				2: SEVERITY.MAJOR,
+			},
+			adjustCount: this.adjustExpectedActionsCount.bind(this),
+		}))
 
-		const lastRequiescat = this.lastRequiescat
+		this.addEventHook({type: 'action', source: this.parser.actor.id, action: this.data.actions.REQUIESCAT.id}, () => this.requiescatUsages.casts++)
+		this.addEventHook({type: 'statusApply', target: this.parser.actor.id, status: this.data.statuses.REQUIESCAT.id}, () => this.requiescatUsages.buffs++)
 
-		// If we're still in the considered window, log our actions to it
-		if (lastRequiescat != null && lastRequiescat.end == null) {
-			lastRequiescat.rotation.push(event)
-		}
+		this.addEvaluator(new RequiescatUsageEvaluator(this.requiescatUsages, this.data.actions.REQUIESCAT.icon))
 	}
 
-	private onApplyRequiescat() {
-		this.castTimeIndex = this.castTime.setInstantCastAdjustment(SPELLS)
-
-		const lastRequiescat = this.lastRequiescat
-
-		if (lastRequiescat != null) {
-			lastRequiescat.hasAssociatedBuff = true
+	private adjustExpectedActionsCount(window: HistoryEntry<EvaluatedAction[]>, action: TrackedAction) {
+		if (window.end == null) {
+			return 0
 		}
-	}
 
-	private onRemoveRequiescat(event: BuffEvent) {
-		if (this.castTimeIndex != null) {
-			this.castTime.reset(this.castTimeIndex)
-			this.castTimeIndex = null
+		const windowDuration = window.end - window.start
+		const expectedGcds = Math.floor(windowDuration / this.globalCooldown.getEstimate()) + 1
+		if (expectedGcds < EXPECTED_HOLY_SPIRITS + 1 && ensureArray(action.action)[0] === this.data.actions.HOLY_SPIRIT) {
+			return (expectedGcds - 1) - EXPECTED_HOLY_SPIRITS
 		}
 
-		const lastRequiescat = this.lastRequiescat
-
-		if (lastRequiescat != null) {
-			lastRequiescat.end = event.timestamp
+		if (expectedGcds === 0 && ensureArray(action.action)[0] === this.data.actions.CONFITEOR) {
```

Comment:
looking at this adjustment function, i'm really starting to question having >1 tracked action per evaulator. it just leads to handling that looks like

```
if it's skill one
	do logic
	return

if it's skill two
	do logic
	return

...
``` 
which seems silly. out of scope for this pr but possibly worth refactoring evaluators down to single responsibility